oci/layout/oci_transport.go

```diff
@@ -182,8 +214,15 @@ func (ref ociReference) getManifestDescriptor() (imgspecv1.Descriptor, error) {
 	if err != nil {
 		return imgspecv1.Descriptor{}, err
 	}
-
 	var d *imgspecv1.Descriptor
+	if ref.sourceIndex < len(index.Manifests) {
```

Comment:
It does the same thing, but I’d much prefer this organized like
```go
if ref.sourceIndex != -1 { // sourceIndex specified
    // validate and use sourceIndex
}
```

(the way I think about it, `sourceIndex` has type “optional `int`”, and `if ref.sourceIndex == -1` (`sourceIndex` has value `nil`/“missing”) it’s not even _valid_ to compute `ref.sourceIndex < …` with a “missing” value.)

[Even efficiency-wise, it’s the same number of comparisons (2) if `sourceIndex` is used; and it’s just one instead of two if it is not used.]

Compare `docker/internal/tarfile.Reader.ChooseManifestItem`.

Reply:
fixed